cpp/src/copying/contiguous_split.cu

```diff
@@ -1158,116 +1273,715 @@ std::vector<packed_table> contiguous_split(cudf::table_view const& input,
     });
 
   // compute total size of each partition
+  // key is the split index
   {
-    // key is split index
     auto keys = cudf::detail::make_counting_transform_iterator(
       0, split_key_functor{static_cast<int>(num_src_bufs)});
     auto values =
       cudf::detail::make_counting_transform_iterator(0, buf_size_functor{d_dst_buf_info});
 
-    thrust::reduce_by_key(rmm::exec_policy(stream),
+    thrust::reduce_by_key(rmm::exec_policy(stream, mr),
                           keys,
                           keys + num_bufs,
                           values,
                           thrust::make_discard_iterator(),
                           d_buf_sizes);
   }
 
-  // compute start offset for each output buffer
+  // compute start offset for each output buffer for each split
   {
     auto keys = cudf::detail::make_counting_transform_iterator(
       0, split_key_functor{static_cast<int>(num_src_bufs)});
     auto values =
       cudf::detail::make_counting_transform_iterator(0, buf_size_functor{d_dst_buf_info});
 
-    thrust::exclusive_scan_by_key(rmm::exec_policy(stream),
+    thrust::exclusive_scan_by_key(rmm::exec_policy(stream, mr),
```

Comment:
Similar question. Probably `thrust` will allocate internal memory for its temporary usage, but since it doesn't return any device memory to us then we don't need to pass `mr`.